Changed inbox behaviour for additional files for scheduled events

Arnei opened the pull request:
Resolves #3649

Previously, additional files for a scheduled event to the inbox would overwrite the previous event or cause an exception in case a workflow was currently running on the scheduled event. Now, new events are created for additional files instead.

Unfortunately I did not find a (reasonably easy) way to number the new events as requested in the linked issue. For now, new events are named after the pattern "originalTitle - newEventId". 

### Your pull request should…

* [x] have a concise title
* [x] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [x] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [x] include migration scripts and documentation, if appropriate
* [x] pass automated tests
* [x] have a clean commit history
* [x] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

Arnei committed:
Changed inbox behaviour for additional files for scheduled events

Previously, additional files for a scheduled event would overwrite the previous files or cause an exception in case a workflow was currently running on the scheduled event. Now, new events are created for additional files instead.

lkiesow commented on modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/Ingestor.java:
```diff
@@ -239,6 +249,31 @@ public RetriableIngestJob call() {
+
+                      // Update dublincore title
+                      try {
+                        String newTitle = mediaPackage.getTitle() + " - " + newId;
```
Some insider knowledge: Never try getting correct metadata from the media package. Always extract it from the Dublin Core catalog. My first test ended up like this:

![Screenshot from 2022-04-20 16-40-36](https://user-images.githubusercontent.com/1008395/164257290-55b03ff8-468b-448f-b319-71a9be897b18.png)

lkiesow replied:
Maybe something like:

```diff
@@ -255,15 +255,17 @@ public class Ingestor implements Runnable {
   try {
     Recording recordingState = schedulerService.getRecordingState(id);
     if (recordingState.getState().equals(UPLOAD_FINISHED)) {
+      var referenceId = mediaPackage.getIdentifier().toString();
       mediaPackage = (MediaPackage) mediaPackage.clone();
-      Id newId = IdImpl.fromUUID();
-      mediaPackage.setIdentifier(newId);
+      mediaPackage.setIdentifier(IdImpl.fromUUID());

       // Update dublincore title
       try {
-        String newTitle = mediaPackage.getTitle() + " - " + newId;
         DublinCoreCatalog dc = DublinCoreUtil.loadEpisodeDublinCore(workspace, mediaPackage).get();
+        var newTitle = dc.get(DublinCore.PROPERTY_TITLE).get(0).getValue()
+            + " (" + Instant.now().getEpochSecond() + ")";
         dc.set(DublinCore.PROPERTY_TITLE, newTitle);
+        dc.set(DublinCore.PROPERTY_REFERENCES, referenceId);
         mediaPackage = updateDublincCoreCatalog(mediaPackage, dc);
         mediaPackage.setTitle(newTitle);
       } catch (Exception e) {
```

This…
- uses the DC title
- adds the original event as reference
- switches from the very long UUID to the ingest time (as unix timestamp) which will at least give you an order for the ingested events

Arnei replied:
Very good suggestions, should be implemented now.

Arnei committed:
Implemented suggestion by @lkiesow

- uses the DC title
- adds the original event as reference
- switches from the very long UUID to the ingest time (as unix timestamp) which will at least give you an order for the ingested events

lkiesow reviewed (approved):
Now this looks better.
![Screenshot from 2022-04-22 13-03-01](https://user-images.githubusercontent.com/1008395/164702644-382a82d5-152e-42cd-9ba7-1e9fada0bb71.png)

ypatios commented:
It seems that the newly created additional events involve a duplication step which copies the initial recording in the new media package. This has as a result that every new part of the same event contains media files of the same flavor (which causes problems in the processing/publishing workflows).
The initial media file(s) should not be included in any of the additional parts. Thus, I regard this as a bug. Right?
Should I reopen the original issue or create a new one?

lkiesow commented:
A second event will inherit all data already in the asset manager for that event.
We could only filter that to e.g. remove media if we wanted to.

Please file a new issue for that. I only noticed it by accident here.

ypatios commented:
Yes, it would make sense to omit the initial recording.
I'll create a new issue. Thanks
